uber/site_sections/registration.py

```diff
@@ -211,21 +211,26 @@ def history(self, session, id):
     @log_pageview
     def watchlist(self, session, attendee_id, watchlist_id=None, message='', **params):
         attendee = session.attendee(attendee_id, allow_invalid=True)
-        if watchlist_id:
-            watchlist_entry = session.watch_list(watchlist_id)
-
-            if 'active' in params:
-                watchlist_entry.active = not watchlist_entry.active
-            if 'confirm' in params:
-                attendee.watchlist_id = watchlist_id
+        if cherrypy.request.method == 'POST':
             if 'ignore' in params:
                 attendee.badge_status = c.COMPLETED_STATUS
-
+            elif watchlist_id:
+                watchlist_entry = session.watch_list(watchlist_id)
+
+                if 'active' in params:
+                    watchlist_entry.active = not watchlist_entry.active
+                    message = 'Watchlist entry updated'
+                if 'confirm' in params:
+                    attendee.watchlist_id = watchlist_id
```

Comment:
I don't really understand what was wrong with the logic from the original method:
```
            if 'active' in params:
                watchlist_entry.active = not watchlist_entry.active
            if 'confirm' in params:
                attendee.watchlist_id = watchlist_id
            if 'ignore' in params:
                attendee.badge_status = c.COMPLETED_STATUS

            session.commit()

            message = 'Watchlist entry updated'
```

It seems like this method could do multiple things at once, right? Also, would adding the watchlist to the attendee via `attendee.watchlist_id = watchlist_id` also be considered "Watchlist entry updated"?

My apologies, I'm not familiar with the watchlist feature at all. The intent of the original code seemed fairly clear though.

Reply:
Only one action will occur per pageload. We do this by giving our form submit buttons themselves names, like 'active' and 'confirm' and 'ignore' -- so the page handler chooses what to do based on which submit button is clicked.

This made sense when all the buttons were part of one form, but now that we've changed the frontend, I think they're all in separate forms. Now the only reason they're grouped together is because they do similar things.

The watchlist entry doesn't care that it has an attendee -- it's a one-to-many relationship. This is so attendees can't dodge bans by registering multiple times. The important part of 'confirming' a watchlist entry is that the attendee now only has one entry, and no others. This is... of dubious use, but in any case, the attendee is the important part of the operation. 😅 

Wait it's like 4:30am, go to bed :C